include/nbl/builtin/glsl/format/encode.glsl

```diff
@@ -82,4 +82,51 @@ uint nbl_glsl_encodeRGB10A2_SNORM(in vec4 col)
 	return quantized.r|quantized.g|quantized.b|quantized.a;
 }
 
+// TODO: break it down into uint nbl_glsl_encode_ufloat_exponent(in float _f32) and nbl_glsl_encode_ufloat_mantissa(in float _f32, in uint mantissaBits, in bool leadingOne)
+uint nbl_glsl_encode_ufloat(in float _f32, in uint mantissaBits)
+{
+	uint minSinglePrecisionVal = floatBitsToUint(nbl_glsl_R11G11B10_MIN);
+	uint maxSinglePrecisionVal = floatBitsToUint(nbl_glsl_R11G11B10_MAX);
+
+	if (_f32 < uintBitsToFloat(maxSinglePrecisionVal))
+	{
+		if (_f32 < uintBitsToFloat(minSinglePrecisionVal))
+			return 0;
+
+		const int exp = int(nbl_glsl_ieee754_extract_biased_exponent(_f32) - 127);
+		const uint mantissa = nbl_glsl_ieee754_extract_mantissa(_f32);
+
+		const uint e = uint(exp + nbl_glsl_R11G11B10_BIAS);
+		const uint m = mantissa >> (23 - mantissaBits);
+		const uint encodedValue = (e << mantissaBits) | m;
+
+		return encodedValue;
+	}
+
+	const uint expMask = ((1 << nbl_glsl_R11G11B10_EXPONENT_BITS) - 1) << mantissaBits;
```

Comment:
turn `((1 << nbl_glsl_R11G11B10_EXPONENT_BITS) - 1) << mantissaBits` into `nbl_glsl_ieee754_compute_exponent_mask(int uint exponentBits, in uint mantissaBits)`

Reply:
Resolved.